test/OpenTelemetry.Exporter.Prometheus.HttpListener.Tests/PrometheusCollectionManagerTests.cs

```diff
@@ -40,9 +40,9 @@ public async Task EnterExitCollectTest()
 
             using (var provider = Sdk.CreateMeterProviderBuilder()
                 .AddMeter(meter.Name)
-#if NETFRAMEWORK
+#if PROMETHEUS_HTTP_LISTENER
                 .AddPrometheusHttpListener()
-#else
+#elif PROMETHEUS_ASPNETCORE
```

Comment:
What if `#else`? Do we really need `#elif PROMETHEUS_ASPNETCORE` here, or `#else` is sufficient?

Reply:
In theory it is sufficient, but for me current version is more clear why there is such configuration. It is unusual to have conditional compilation, especially in tests.

Reply:
Personal opinion. I find having `#elif PROMETHEUS_ASPNETCORE` more readable.

Reply:
I'm concerned about the future maintenance. For example, what if someone added yet another Prometheus exporter (e.g. Prometheus push gateway exporter)? - would this test case automatically pass or generating some confusing error? Maybe having something explicit would help:

```csharp
# if HttpListener
do something
#elif ASP.NET Core
do something else
#else
raise a compile/build time error so folks know something screwed up
```

Reply:
It's not a blocker though, can be addressed later.

Reply:
If you apply this test to the scenario without any Prometheus exporter it fails around line 52 while after try finding exporter.

BTW I have opened this branch for changes for approvers. Feel free to apply changes if needed 🙂